airbyte-integrations/connectors/source-marketo/source_marketo/source.py

```diff
@@ -328,13 +328,61 @@ class Leads(MarketoExportBase):
 
     cursor_field = "updatedAt"
 
-    def __init__(self, config: Mapping[str, Any]):
+    def __init__(self, config: Mapping[str, Any], ):
         super().__init__(config, self.name)
 
     @property
     def stream_fields(self):
-        return list(self.get_json_schema()["properties"].keys())
+        leads_schema_fields = list(self.get_json_schema()["properties"].keys())
+        available_lead_fields = self.describe_leads_fields()
+        if available_lead_fields is not None:
+            return [
+                schema_field
+                for schema_field in leads_schema_fields
+                if schema_field in available_lead_fields
+            ]
+        else:
```

Comment:
Is this really the right thing to do if we get a server error when calling describe?

Reply:
I'm very lukewarm about the error handling here, but it's consistent with how the rest of their new code works: basically no hard-failing on anything. I personally would prefer something like how we do stuff in our validation: run each module, collect all errors, then throw only at the very end.